idb/postgres/postgres.go

```diff
@@ -1658,9 +1672,15 @@ func (db *IndexerDb) GetAccounts(ctx context.Context, opts idb.AccountQueryOptio
 	round, err := db.getMaxRoundAccounted(ctx, tx)
 	if err != nil {
 		err = fmt.Errorf("account round err %v", err)
+		// Because we return a channel into a "callWithTimeout" function,
+		// We need to make sure that rollback is called before close()
+		// otherwise we can end up with a situation where "callWithTimeout"
+		// will cancel our context, resulting in connection pool churn
 		out <- idb.AccountRow{Error: err}
```

Comment:
This should be below `Rollback()`. There are are many similar cases below.

Reply:
Didn't you say earlier that it should be above AccountRow?

Reply:
First we should rollback and then write an error to the channel and close the channel.